codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -164,4 +183,51 @@ class FileCodesignVisitor {
       processManager: processManager,
     );
   }
+
+  /// Visit and codesign a binary with / without entilement.
+  ///
+  /// At this stage, the virtual [entitlementCurrentPath] accumulated through the recursive visit, is compared
+  /// against the paths extracted from [fileWithEntitlements], to help determine if this file should be signed
+  /// with entitlements.
+  Future<void> visitBinaryFile({required FileSystemEntity binaryEntity, required String entitlementParentPath}) async {
+    if (await isSymlink(binaryEntity.absolute.path, processManager)) {
+      return;
+    }
+
+    final String currentFileName = binaryEntity.absolute.path.split('/').last;
```

Comment:
use `.basename` https://pub.dev/documentation/file/latest/file/FileSystemEntity/basename.html

Reply:
Good point, thanks for pointing out!